gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/editor/bottomsheet/GravatarQuickEditorBottomSheet.kt

```diff
@@ -132,53 +137,74 @@ private fun GravatarModalBottomSheet(
         }
     }
 
-    GravatarTheme {
-        ModalBottomSheet(
-            state = modalBottomSheetState,
-        ) {
-            Scrim(
-                modifier = Modifier.clickable { modalBottomSheetState.currentDetent = Hidden },
-                scrimColor = MaterialTheme.colorScheme.scrim.copy(alpha = 0.32f),
-            )
-            Sheet(
-                modifier = Modifier
-                    .clip(RoundedCornerShape(topStart = 28.dp, topEnd = 28.dp))
-                    .background(MaterialTheme.colorScheme.surfaceColorAtElevation(1.dp))
-                    .widthIn(max = 640.dp)
-                    .fillMaxWidth()
-                    .padding(
-                        WindowInsets.navigationBars
-                            .only(WindowInsetsSides.Vertical)
-                            .asPaddingValues(),
-                    ),
+    val configuration = Configuration(LocalConfiguration.current).apply {
+        uiMode = when (colorScheme) {
+            GravatarColorScheme.LIGHT -> Configuration.UI_MODE_NIGHT_NO
+            GravatarColorScheme.DARK -> Configuration.UI_MODE_NIGHT_YES
+            GravatarColorScheme.SYSTEM -> uiMode
+        }
+    }
+    CompositionLocalProvider(
+        LocalConfiguration provides configuration,
+    ) {
+        GravatarTheme {
+            ModalBottomSheet(
+                state = modalBottomSheetState,
             ) {
-                val window = LocalModalWindow.current
-                val isDarkTheme = isSystemInDarkTheme()
-                LaunchedEffect(Unit) {
-                    window.navigationBarColor = Color.TRANSPARENT
-                    WindowInsetsControllerCompat(window, window.decorView).isAppearanceLightNavigationBars =
-                        !isDarkTheme
-                }
-                Surface(
-                    modifier = Modifier
-                        .fillMaxWidth(),
-                    tonalElevation = 1.dp,
+                // Modal content must be taking the uiMode from Activity and doesn't respect
+                // the above set CompositionLocalProvider
+                CompositionLocalProvider(
```

Comment:
I've been investigating why we need to add this again here. 

The [ModalBottomSheet](https://github.com/composablehorizons/compose-unstyled/blob/c62bab5babdabceb634ec66de05f5370c161b66d/core/src/commonMain/kotlin/ModalBottomSheet.kt#L171) pass the content to another composable; [Modal](https://github.com/composablehorizons/compose-unstyled/blob/c62bab5babdabceb634ec66de05f5370c161b66d/core/src/androidMain/kotlin/Modal.android.kt#L40) which is using the legacy view system under the hood. 

I think we are losing the configuration that was already provided at that point. That is why we need to set the `LocalConfiguration` inside the `ModalBottomSheet` `content` too.

Reply:
Also, mention that this is a pain because it forces us to add the `colorScheme: GravatarColorScheme` param to the composable instead of providing a `LocalConfiguration` as third-party apps can do to override the UI components style.

Reply:
Yes, that is true. The thing I don't get is why I need the outer `CompositionLocalProvider`, because technically, all the content is wrapped with the inner one already.

Reply:
Yeah, same here. I don't understand it.